cmd/snap/cmd_snap_op.go

```diff
@@ -252,10 +252,25 @@ func (mx *channelMixin) setChannelFromCommandline() error {
 		mx.Channel = ch.chName
 	}
 
-	if !strings.Contains(mx.Channel, "/") && mx.Channel != "" && mx.Channel != "edge" && mx.Channel != "beta" && mx.Channel != "candidate" && mx.Channel != "stable" {
-		// shortcut to jump to a different track, e.g.
-		// snap install foo --channel=3.4 # implies 3.4/stable
-		mx.Channel += "/stable"
+	if mx.Channel != "" {
+		if ch, err := channel.Parse(mx.Channel, ""); err != nil {
+			full, er := channel.Full(mx.Channel)
+			if er != nil {
+				// the parse error has more detailed info
+				return err
+			}
+
+			// TODO: get escapes in here so we can bold the Warning
+			head := i18n.G("Warning:")
+			msg := i18n.G("Specifying a channel %q is relying on undefined behaviour. Interpreting it as %q for now, but this will be an error later.\n")
```

Comment:
As discussed on IRC, probably needs to tweak to use the US spelling.

Reply:
I'll wait on @degville's feedback before butchering it :-) thanks for the review

Reply:
Sounds like tweaking the string could be a followup? Would be nice to merge the rest, especially since this is green now :)

Reply:
Sounds good to me. We can always fix it later.